diff --git a/include/sha1.h b/include/sha1.h
index 734d1fb..6ad159d 100644
--- a/include/sha1.h
+++ b/include/sha1.h
@@ -45,6 +45,8 @@ typedef struct
 }
 sha1_context;
 
+typedef sha1_context SHA1_CTX;
+
 /**
  * \brief	   SHA-1 context setup
  *
@@ -59,7 +61,7 @@ void sha1_starts( sha1_context *ctx );
  * \param input    buffer holding the  data
  * \param ilen	   length of the input data
  */
-void sha1_update( sha1_context *ctx, unsigned char *input, int ilen );
+void sha1_update( sha1_context *ctx, const unsigned char *input, int ilen );
 
 /**
  * \brief	   SHA-1 final digest
@@ -76,7 +78,7 @@ void sha1_finish( sha1_context *ctx, unsigned char output[20] );
  * \param ilen	   length of the input data
  * \param output   SHA-1 checksum result
  */
-void sha1_csum( unsigned char *input, int ilen,
+void sha1_csum( const unsigned char *input, int ilen,
 		unsigned char output[20] );
 
 /**
@@ -87,7 +89,7 @@ void sha1_csum( unsigned char *input, int ilen,
  * \param output   SHA-1 checksum result
  * \param chunk_sz watchdog triggering period (in bytes of input processed)
  */
-void sha1_csum_wd (unsigned char *input, int ilen,
+void sha1_csum_wd (const unsigned char *input, int ilen,
 		unsigned char output[20], unsigned int chunk_sz);
 
 /**
@@ -108,10 +110,14 @@ int sha1_file( char *path, unsigned char output[20] );
  * \param ilen	   length of the input data
  * \param output   HMAC-SHA-1 result
  */
-void sha1_hmac( unsigned char *key, int keylen,
-		unsigned char *input, int ilen,
+void sha1_hmac( const unsigned char *key, int keylen,
+		const unsigned char *input, int ilen,
 		unsigned char output[20] );
 
+void hmac_starts( sha1_context *ctx, const unsigned char *key, unsigned int len);
+void hmac_update( sha1_context *ctx, const unsigned char *data, unsigned int len);
+void hmac_finish( sha1_context *ctx, const unsigned char *key, unsigned int len, unsigned char *output);
+
 /**
  * \brief	   Checkup routine
  *
